### ISSUE TITLE
Readme recommends melpa

### ISSUE THREAD

**Issue by technomancy** [technomancy/find-file-in-project#53]
It makes me uncomfortable that the readme here recommends the use of melpa.

There are a number of subtle gotchas that a user is likely to get bitten by if they add melpa to their package listing; including getting unstable versions of packages that they intended to install from another source. (Switching to recommending melpa-stable would solve this problem.)

But beyond that, given that it is basically impossible to securely connect over TLS from within Emacs, I don't feel comfortable recommending package.el usage for a project that I'm associated with: https://glyph.twistedmatrix.com/2015/11/editor-malware.html

**Comment by redguardtoo** [comment]
Make sense to use melpa-stable.

As for the security, there is some complexity. Other package like `elpy` is dependent on find-file-in-project. So python developers will install find-file-in-project through melpa anyway.

If you don't recommend them install find-file-in-project through melpa, version conflict will happen.

That issue actually bite me. see https://github.com/redguardtoo/emacs.d/commit/18ac5a1

**Comment by technomancy** [comment]
I'm not saying that it shouldn't be possible to install through melpa;
obviously users who pull it in transitively will use whatever system it
is that the other library recommends. While it is regrettable that
others don't care about or understand the problems, it's not my concern.

I'm just saying it feels irresponsible to recommend this installation
mechanism when it's known to be so flawed that it can be easily
exploited by an unsophisticated attacker. Originally I was going to say
we should just document how to install from git or wget, but now I see
that a dependency has been added which makes that difficult. But el-get
does support installing over git https, which can be done securely.

**Comment by technomancy** [comment]
I noticed that the usage of ivy is limited to a single call, and that we
already have fallback code for using ido instead. I pushed a change
that makes it so that it will still work even when ido is not installed,
so now we can document a dependency-free way of using find-file-in-project.

I don't mind also including instructions for melpa as long as the
security concerns are also mentioned. Hopefully Emacs 25 will allow
people to use package.el with TLS safely, so these problems may not be a
deal-breaker in the future, but people should know of the risks when
they are using 24.

**Comment by redguardtoo** [comment]
agree, we should warn the user the security risk using melpa on emacs24.

**Comment by unhammer** [comment]
Weren't (some of) the TLS issues solved in Emacs 25 though? `/usr/share/emacs/25.1/etc/NEWS` says 
```
** TLS
*** Fatal TLS errors are now silent by default.

*** If Emacs isn't built with TLS support, an external TLS-capable
program is used instead.  This program used to be run in --insecure
mode by default, but has now changed to be secure instead, and will
fail if you try to connect to non-verifiable hosts.  This is
controlled by the 'tls-program' variable.
```

**Comment by redguardtoo** [comment]
Looks good. Though I'm not familiar this thing.
